src/info/info_redis/types/blocked_queries.c

```diff
@@ -49,10 +73,15 @@ BlockedCursorNode* BlockedQueries_AddCursor(BlockedQueries* blockedQueries, Weak
 }
 
 void BlockedQueries_RemoveQuery(BlockedQueryNode* blockedQueryNode) {
-  StrongRef_Release(blockedQueryNode->spec);
+  if (blockedQueryNode->spec.rm) {
+    StrongRef_Release(blockedQueryNode->spec);
+  }
   dllist_delete(&blockedQueryNode->llnode);
 }
 
 void BlockedQueries_RemoveCursor(BlockedCursorNode* blockedCursorNode) {
+  if (blockedCursorNode->spec.rm) {
+    StrongRef_Release(blockedCursorNode->spec);
```

Comment:
I think it’s better to have the bug fix on its own, and CP it to where we need it.
Also, we should add a mechanism on sanitizer shutdown to make sure we don’t leak indexes